### PR TITLE
Bug/issue 67 make sls work with pnpm

### LINKED ISSUE #67: nx-serverless executors don't work with pnpm
I'm using pnpm and the plugin crashes when running executors because of the specific use of npx:

https://github.com/Bielik20/nx-plugins/blob/master/packages/nx-serverless/src/executors/sls/executor.ts#L21

https://www.npmjs.com/package/detect-package-manager can help.

I'll put up a PR, but I don't know how to test plugins.

### PR DESCRIPTION
**Pull request by ALMaclaine**
fixes #67

### CODE REVIEW

File: packages/nx-serverless/src/utils/get-sls-command.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  const packageManager = detectPackageManager();
+  switch (packageManager) {
+    case 'yarn':
+      return 'npx sls';
+    case 'pnpm':
+      return 'sls';
+    case 'npm':
+    default:
+      return 'npx sls';
+  }
```
Comment (Bielik20) [inline packages/nx-serverless/src/utils/get-sls-command.ts:13]:
```suggestion
  switch (packageManager) {
    case 'pnpm':
      return 'sls';
    case 'yarn':
    case 'npm':
    default:
      return 'npx sls';
  }
```

**Review by Bielik20 — COMMENTED** [review]
Great 👍 Let's squash those commits before merge

**Comment by ALMaclaine** [comment]
think I've fixed it

**Comment by Bielik20** [comment]
Tests are still failing :/

**Comment by ALMaclaine** [comment]
@Bielik20 Do you know whats wrong because I don't.

**Comment by Bielik20** [comment]
No idea. In my opinion it should work just fine. Maybe I will find some time to debug it. I usually run those commands locally and inspect logs.

**Comment by ALMaclaine** [comment]
I copy pasted the build directory in like you suggested doing and it worked in my project.

**Comment by ALMaclaine** [comment]
@Bielik20 these fail on master also:

```sh
nx-plugins git:(master) nx e2e nx-serverless-e2e

> nx run nx-serverless-e2e:e2e


 >  NX   Running target build for project nx-serverless and 1 task(s) it depends on

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

> nx run nx-serverless:build

Compiling TypeScript files for project "nx-serverless"...
Done compiling TypeScript files for project "nx-serverless".

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Successfully ran target build for project nx-serverless

   Nx read the output from the cache instead of running the command for 1 out of 2 tasks.

 FAIL   nx-serverless-e2e  e2e/nx-serverless-e2e/tests/serverless.spec.ts (120.193 s)
  serverless e2e
    serverless-bundle
      ✓ should create serverless (50287 ms)
    @ns3/nx-serverless/plugin
      ✕ should create serverless (22852 ms)
    --directory and --tags
      ✕ should create src in the specified directory with tags (1099 ms)

  ● serverless e2e › @ns3/nx-serverless/plugin › should create serverless

    Command failed: yarn nx package nx-serverless8735837
    error Command failed with exit code 1.



  ● serverless e2e › --directory and --tags › should create src in the specified directory with tags

    Command failed: yarn nx generate @ns3/nx-serverless:application serverless485392 --directory subdir --tags e2etag,e2ePackage
    error Command failed with exit code 1.



Test Suites: 1 failed, 1 total
Tests:       2 failed, 1 passed, 3 total
Snapshots:   0 total
Time:        121.088 s
Ran all test suites.

 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target e2e for project nx-serverless-e2e (2m)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
 
   View structured, searchable error logs at https://nx.app/runs/ziB1BPfEL4
```

**Comment by Bielik20** [comment]
I have updated dependencies, it should work fine now 🤞 Rebase with current master.

And while doing so I have found that Nx exposes `getPackageManagerCommand`, maybe we could use that instead :D

**Comment by ALMaclaine** [comment]
I dont think getPackageManagerCommand is correct for this scenario, here is it's interface:

```
export interface PackageManagerCommands {
    install: string;
    ciInstall: string;
    add: string;
    addDev: string;
    rm: string;
    exec: string;
    list: string;
    run: (script: string, args: string) => string;
}
```

**Comment by Bielik20** [comment]
Yeah, and I think `exec` is what we want, or at leat it should be the thing. 

```ts
const { exec } = getPackageManagerCommand();
fullCommand = `${exec} sls ${command} ${stringifiedArgs}`.trim();
```

So, it should be what we want but I am still not sure it will work with `pnpm`, but it should.

**Comment by ALMaclaine** [comment]
@Bielik20 this seems to be working and tests are fixed. Seems like something you can adjust in the future if you think it's necessary.

**Pull request merged**